Yubico.YubiKey/src/Yubico/YubiKey/Scp/SessionKeys.cs

```diff
@@ -0,0 +1,108 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Security.Cryptography;
+
+namespace Yubico.YubiKey.Scp
+{
+    internal class SessionKeys : IDisposable
+    {
+        /// <summary>
+        /// Gets the session MAC key.
+        /// </summary>
+        public ReadOnlyMemory<byte> MacKey => _macKey;
+
+        /// <summary>
+        /// Gets the session encryption key.    
+        /// </summary>
+        public ReadOnlyMemory<byte> EncKey => _encryptionKey;
+
+        /// <summary>
+        /// Gets the session RMAC key.
+        /// </summary>
+        public ReadOnlyMemory<byte> RmacKey => _rmacKey;
+
+        /// <summary>
+        /// Gets the session data encryption key (DEK).
+        /// </summary>
+        public ReadOnlyMemory<byte>? DataEncryptionKey => _dataEncryptionKey;
+
+        private const byte KeySizeBytes = 16;
+        private readonly Memory<byte> _macKey = new byte[KeySizeBytes];
+        private readonly Memory<byte> _encryptionKey = new byte[KeySizeBytes];
+        private readonly Memory<byte> _rmacKey = new byte[KeySizeBytes];
+        private readonly Memory<byte> _dataEncryptionKey = new byte[KeySizeBytes];
+        private bool _disposed;
+
+        /// <summary>
+        /// Session keys for Secure Channel Protocol (SCP).
+        /// </summary>
+        /// <param name="macKey">The session MAC key.</param>
+        /// <param name="encryptionKey">The session encryption key.</param>
+        /// <param name="rmacKey">The session RMAC key.</param>
+        /// <param name="dataEncryptionKey">The session data encryption key. Optional.</param>
+        public SessionKeys(
+            ReadOnlyMemory<byte> macKey,
+            ReadOnlyMemory<byte> encryptionKey,
+            ReadOnlyMemory<byte> rmacKey,
+            ReadOnlyMemory<byte> dataEncryptionKey)
+        {
+            ValidateKeyLength(macKey, nameof(macKey));
+            ValidateKeyLength(encryptionKey, nameof(encryptionKey));
+            ValidateKeyLength(rmacKey, nameof(rmacKey));
+            ValidateKeyLength(dataEncryptionKey, nameof(dataEncryptionKey));
+            
+            macKey.CopyTo(_macKey);
+            encryptionKey.CopyTo(_encryptionKey);
+            rmacKey.CopyTo(_rmacKey);
+            dataEncryptionKey.CopyTo(_dataEncryptionKey);
+        }
+        
+        private static void ValidateKeyLength(ReadOnlyMemory<byte> key, string paramName)
+        {
+            if (key.Length != KeySizeBytes)
+            {
+                throw new ArgumentException("Incorrect session key length. Must be 16.", paramName);
+            }
+        }
+
+        public void Dispose()
+        {
+            Dispose(disposing: true);
```

Comment:
suggestion for backlog - unify the way how this method is called throughout the SDK.

`Dispose(disposing: value)` vs `Dispose(value)`

Reply:
Theres a lot of unifying to do. Well probably save it for a 2.0 release that harmonizes the different SDKs